cmd/zpool/zpool_main.c

```diff
@@ -1903,6 +1903,21 @@ print_status_config(zpool_handle_t *zhp, status_cbdata_t *cb, const char *name,
 
 		vname = zpool_vdev_name(g_zfs, zhp, child[c],
 		    cb->cb_name_flags | VDEV_NAME_TYPE_ID);
+
+		/*
+		 * print_status_config() want to print all vdevs statuses
+		 * independently of each other. However, only the root vdev has
+		 * ZPOOL_CONFIG_SCAN_STATS, which we need on the child vdevs to
+		 * determine whether or not to print "(repairing)" and
+		 * "(resilvering)".  Copy ZPOOL_CONFIG_SCAN_STATS to each child
+		 * vdev to make it work.
+		 */
+		if (ps != NULL) {
+			fnvlist_add_uint64_array(child[c],
```

Comment:
Instead of mutating the nvlist of the child could we replace this:
```
	(void) nvlist_lookup_uint64_array(nv, ZPOOL_CONFIG_SCAN_STATS,
	    (uint64_t **)&ps, &c);

	if (ps != NULL && ps->pss_state == DSS_SCANNING &&
	    vs->vs_scan_processed != 0 && children == 0) {
		(void) printf(gettext("  (%s)"),
		    (ps->pss_func == POOL_SCAN_RESILVER) ?
		    "resilvering" : "repairing");
	}
```
With something like this?
```
nvlist_t *nvroot = fnvlist_lookup_nvlist(zpool_get_config(zhp, NULL),
    ZPOOL_CONFIG_VDEV_TREE);
verify(nvlist_lookup_uint64_array(nvroot, ZPOOL_CONFIG_VDEV_STATS,
    (uint64_t **)&vs, &c) == 0);

(void) nvlist_lookup_uint64_array(nvroot,
    ZPOOL_CONFIG_SCAN_STATS, (uint64_t **)&ps, &c);

if (ps != NULL && ps->pss_state == DSS_SCANNING &&
    vs->vs_scan_processed != 0 && children == 0) {
	(void) printf(gettext("  (%s)"),
	    (ps->pss_func == POOL_SCAN_RESILVER) ?
	    "resilvering" : "repairing");
}
```

Reply:
Yea, that's way better.  I fixed it in my latest push